### ISSUE TITLE
[plugin] Firebase message notification with actions

### ISSUE THREAD

**Issue by jaumard** [flutter/flutter#26358]
I need to add some actions in my notifications (like `reply`, `delete`, `mark as read`...). I wonder how it's possible with this plugin ? 

From what I know on iOS you need to set the category under `click_action` but from the doc of the plugin `click_action` need to be set to `FLUTTER_NOTIFICATION_CLICK` so impossible to do more than one category. 

For Android, there no way to add actions on our notification because the plugin is showing the notification by itself and from what I know we can't hook those to add our actions.

So it would be nice if we can put whatever we want on `click_action` like this we can set category for iOS. 

On Android side, would be nice to be able to set all possible values of our `click_action` on the manifest like we have to do with `FLUTTER_NOTIFICATION_CLICK` and also notify the plugin that those values need to be used like a notification click. 
We also need a hook on the plugin to add one or more actions to the notification that will be added. 

Of course it will force us to do native code, but for this I don't see any other options. For now the plugin only allow you to do really basic notifications, it should be more flexible.

**Comment by zoechi** [comment]
I think this is a duplicate of #22072 
This way you could pass data like "action" with the notification and then execute code accordingly.
Currently the app gets only such notification data when the app is in the foreground (related #21043).

**Comment by jaumard** [comment]
Definitely not a duplicate of #22072 because on this one they want to execute dart code in background, here I just want to add actions to my notification like this for example (share, call):
![screen shot 2016-06-08 at 11 32 25 am](https://user-images.githubusercontent.com/2970449/50964659-20f21580-14d0-11e9-9c21-b02018e93dae.png)
So nothing related. 

It's somehow related to #21043 but my purpose is to enhance the notification on native side not retrieve information about the notification on Dart side like it's about on this issue, it's possible native side because even in background the app is responsible for showing notification (at least on Android, for iOS it uses category which we can't use because of `FLUTTER_NOTIFICATION_CLICK `). 

For me this issue is still legit :)

**Comment by zoechi** [comment]
With #22072 you might be able to use local_notifications and show custom notifications using Flutter code.

I'll leave it to the plugin maintainers.

**Comment by jaumard** [comment]
You'll be able to do it on Android, but not on iOS as it's the system who show the notification and not the app. But let's see :)

**Comment by AndruByrne** [comment]
@zoechi, flutter_local_notifications is [incompatible with flutter_messaging at this time](https://github.com/flutter/flutter/issues/22099); did you actually mean the deprecated plugin local_notifications? Because if so, that could be a clue to a solution not blocked by Add2App compatibility.

**Comment by rolandbodenstaff** [comment]
This would be a very useful feature indeed

**Comment by Fernandomr88** [comment]
Any news on this? 🤔

**Comment by kroikie** [comment]
@jaumard

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/514. Any further collaboration will be done there.
